python/ray/tune/integration/comet.py

```diff
@@ -199,6 +209,33 @@ def log_trial_result(self, iteration: int, trial: "Trial", result: Dict):
         for k, v in episode_logs.items():
             experiment.log_curve(k, x=range(len(v)), y=v, step=step)
 
+    def log_trial_save(self, trial: "Trial"):
+        if self.save_checkpoints and trial.checkpoint:
+            if trial not in self._trial_experiments:
+                self.log_trial_start(trial)
+
+            experiment = self._trial_experiments[trial]
+
+            artifact = comet_ml.Artifact(
+                name=f"checkpoint_{(str(trial))}", artifact_type="model"
+            )
+
+            # Walk through checkpoint directory and add all files to artifact
+            checkpoint_root = trial.checkpoint.value
+            for root, dirs, files in os.walk(checkpoint_root):
+                rel_root = os.path.relpath(root, checkpoint_root)
+                for file in files:
+                    local_file = os.path.join(checkpoint_root, rel_root, file)
+                    logical_path = os.path.join(rel_root, file)
+
+                    # Strp leading `./`
+                    if logical_path.startswith("./"):
+                        logical_path = logical_path[2:]
+
+                    artifact.add(local_file, logical_path=logical_path)
```

Comment:
Oh man, does Comet have no support for logging directories as artifacts?

Reply:
Unfortunately not...